.github/workflows/build-test.yml

```diff
@@ -6,9 +6,12 @@ on:
   pull_request:
 
 jobs:
-  build-lint-test:
-    name: Build, Lint, and Test
+  prepare:
+    name: Prepare
     runs-on: ubuntu-20.04
+    outputs:
+      YARN_CACHE_DIR: ${{ steps.yarn-cache-dir.outputs.YARN_CACHE_DIR }}
+      YARN_VERSION: ${{ steps.yarn-version.outputs.YARN_VERSION }}
     strategy:
       matrix:
         node-version: [14.x, 16.x, 18.x]
```

Comment:
Rather than use a matrix for each job, maybe it would be better to create a reusable "build-lint-test" workflow and invoke it using a matrix?

Right now, all runs have to wait for all three prepare runs to complete. But if we move this into a reusable workflow, we can start running the build/test/lint steps for the first `prepare` job to complete without waiting for the others. And I don't think it would cost any additional overhead either, but I'm not sure.